core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -199,21 +198,29 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
         });
   }
 
-  private void updateCatalogTables(Connection connection) throws SQLException {
-    DatabaseMetaData dbMeta = connection.getMetaData();
-    ResultSet typeColumn =
-        dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
-    if (typeColumn.next()) {
-      LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
-    } else {
-      if (PropertyUtil.propertyAsBoolean(
-          catalogProperties, JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY, false)) {
-        connection.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
-      } else {
-        LOG.warn(VIEW_WARNING_LOG_MESSAGE);
-        schemaVersion = JdbcUtil.SchemaVersion.V0;
-      }
-    }
+  private void checkAndMaybeUpdateSchema() throws SQLException, InterruptedException {
+    LOG.trace(
+        "Checking the SQL schema version and eventually update if {} is true",
+        JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY);
+    connections.run(
+        conn -> {
+          DatabaseMetaData dbMeta = conn.getMetaData();
+          ResultSet typeColumn =
+              dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
+          if (typeColumn.next()) {
+            LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
+            return true;
+          } else {
+            if (PropertyUtil.propertyAsBoolean(
+                catalogProperties, JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY, false)) {
+              return conn.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
+            } else {
+              LOG.warn(VIEW_WARNING_LOG_MESSAGE);
+              schemaVersion = JdbcUtil.SchemaVersion.V0;
```

Comment:
Instead of downgrading the schema, I think it should start at `V0` and be upgraded if needed. That's more conservative, in case there are any more unintended uses that don't run the check.

Reply:
In that case it would mean that we should init the schema with `V0` when initialize is `true` and update in second step, right ?

Reply:
Yes. I think we should init with the most conservative option and upgrade it iff we detect that the later version can be used.

Reply:
That makes sense. I will add the SQL statement to create the `V0` schema and eventually update in a second step. Thanks !